src/test/java/tests/ConfigFileTest.java

```diff
@@ -0,0 +1,208 @@
+package tests;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+import prefs.Config;
+import prefs.ConfigFile;
+import prefs.SessionConfig;
+import prefs.Preferences;
+import ui.TestController;
+
+
+import java.io.*;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConfigFileTest {
+
+    private static final String CHARSET = "UTF-8";
+
+    private static final String HUBTURBO_REPO_ID = "HubTurbo/HubTurbo";
+    private static final String HUBTURBO_REPO_ALIAS = "ht";
+
+    private static final String POWERPOINTLABS_REPO_ID = "PowerPointLabs/PowerPointLabs";
+    private static final String POWERPOINTLABS_REPO_ALIAS = "ppt";
+
+    private static final String DUMMY_REPO_ID = "Dummy/Dummy";
+    private static final String DUMMY_REPO_ALIAS = "dy";
+
+    private static final String DUMMY_CONFIG_DIRECTORY = Preferences.DIRECTORY;
+    private static final String DUMMY_CONFIG_FILENAME = "dummyConfig.json";
+    private static final String SESSION_CONFIG_DIRECTORY = Preferences.DIRECTORY;
+    private static final String TEST_SESSION_CONFIG_FILENAME = TestController.TEST_SESSION_CONFIG_FILENAME;
+
+    private static final String DUMMY_CONFIG_FILE_CONTENTS_NO_WHITESPACE =
+            String.format("{\"repoAliases\":{\"%s\":\"%s\",\"%s\":\"%s\",\"%s\":\"%s\"}}",
+                    HUBTURBO_REPO_ID,
+                    HUBTURBO_REPO_ALIAS,
+                    POWERPOINTLABS_REPO_ID,
+                    POWERPOINTLABS_REPO_ALIAS,
+                    DUMMY_REPO_ID,
+                    DUMMY_REPO_ALIAS);
+
+    private static final String DUMMY_BOARD_NAME = "DummyBoardName";
+
+    /**
+     * Test if the dummy config class can write its config information to a file
+     */
+    @Test
+    public void dummyConfigImplementation_serialisation_ableToWriteFile() {
```

Comment:
The naming convention is (note that 2nd and 3rd parts are Pascal case so that the whole thing remains camelCase):

**`MethodName_StateUnderTest_ExpectedBehavior`** 

Following are some of the example:
`isAdult_AgeLessThan18_False`
`withdrawMoney_InvalidAccount_ExceptionThrown`
`admitStudent_MissingMandatoryFields_FailToAdmit`